### PR TITLE
Update docker dir doc and Dockerfile

### PR DESCRIPTION
**Pull request by kimlisa**
#### Reason:
- The inconsistent tag name from `teleport-buildbox` fails to build teleport cluster

#### Changes made to remove error:
- Edit docker/Dockerfile `teleport-buildbox` tag from `latest` to `go1.13.2` because running `make docker` command from root, produces a `teleport-buildbox` image tagged with `go1.13.2`.

### CODE REVIEW

**Review by webvictim — APPROVED** [review]

**Review by benarent — APPROVED** [review]

File: docker/README.md [hunk 1]
```diff
@@ -5,15 +5,17 @@ for testing & development purposes.
 
 ### Building 
 
-First, you need to build `teleport:latest` Docker image. This image is built
-when you type `make build` BUT...
+First, you need to build `teleport:latest` Docker image.
```
Comment (benarent) [inline docker/README.md]: Do we need `teleport:latest` or `FROM teleport-buildbox:go1.13.2 `

File: docker/README.md [hunk 2]
```diff
@@ -5,15 +5,17 @@ for testing & development purposes.
...
 
-But you have to build the base image first, by running `make docker`
-from `$GOPATH/github.com/gravitational/teleport` (repository base dir).
+Run the following commands from `$GOPATH/github.com/gravitational/teleport` (repository base dir):
 
-### Starting 
+```bash
```
Comment (benarent) [inline docker/README.md]: Can you also update the main Readme.md with these changes https://github.com/gravitational/teleport#building-teleport

File: docker/README.md (hunk 1 above)
Comment (kimlisa) [reply docker/README.md]: I think it is `teleport:latest`?:

Screenshot from 2020-03-03 14-05-59

File: docker/README.md (hunk 2 above)
Comment (kimlisa) [reply docker/README.md]: Added a docker section to root README: https://github.com/gravitational/teleport/pull/3400/commits/c22eec54c5762a118687c3a633c3473f472b967b

File: README.md [hunk 3]
```diff
@@ -54,15 +54,20 @@ $ sudo teleport start
 In a production environment Teleport must run as root. But to play, just do `chown $USER /var/lib/teleport` 
 and run it under `$USER`, in this case you will not be able to login as someone else though.
 
-If you wish to deploy Teleport inside a Docker container:
+## Docker
 
+### Deploy Teleport
```
Comment (kimlisa) [inline README.md]: @benarent I am in the middle of getting answers to get this working, so I will keep this as is for now

**Comment by kimlisa** [comment]
@russjones I am unsure why jenkin is failing?

**Comment by russjones** [comment]
@kimlisa It looks like its our usual flaky tests. Leave a comment "@gravitational-jenkins retest this please" to re-run tests.

**Comment by benarent** [comment]
retest this please

File: docker/README.md (hunk 1 above)
Comment (webvictim) [reply docker/README.md]: Yes, it is. It's a local tag rather than anything pushed elsewhere so it doesn't matter too much.

**Pull request merged**